src/pages/signup/SignupForm.tsx

```diff
@@ -0,0 +1,270 @@
+import { Field, Formik, FormikActions } from 'formik';
+import * as React from 'react';
+import { Button, Col, FormFeedback, FormText, Form, FormGroup, Input, Label, Row } from 'reactstrap';
+import { compose, graphql } from 'react-apollo';
+import * as Yup from 'yup';
+
+import { CreateUser, CREATE_OR_LOGIN_WITH_PROVIDERS, CREATE_USER, User } from 'networking/users';
+import { auth, login, signInWithGooglePopUp } from 'utils/firebase';
+import { getFriendlyErrorMessage } from 'utils/validators';
+
+interface SignupProps {
+  createOrLoginWithProviders: (id: string) => Promise<any>;
+  createUser: (user: CreateUser) => Promise<User>;
+}
+
+const LoginSchema = Yup.object().shape({
+  signupEmail: Yup.string()
+    .email('Please enter a valid email address.')
+    .required('Please enter an email address.'),
+  signupFirstName: Yup.string()
+    .required('Please enter your first name.'),
+  signupLastName: Yup.string()
+    .required('Please enter your last name.'),
+  signupPassword: Yup.string()
+    .min(8, 'Your password is too short, please enter a password that is at least 8 characters long')
+    .required('Please enter a valid password.'),
+  signupConfirmPassword: Yup.string()
+    .oneOf([Yup.ref('signupPassword')], 'Your passwords do not match.')
+    .required('Please re-enter your password here.'),
+});
+
+interface SignupFormInput {
+  signupEmail: string;
+  signupFirstName: string;
+  signupLastName: string;
+  signupPassword: string;
+  signupConfirmPassword: string;
+  authenticationError: string;
+}
+
+interface State {
+  providerError: string | null;
+}
+
+class SignupForm extends React.Component<SignupProps, State> {
+  readonly state = {
+    providerError: null,
+  };
+
+  render() {
+    return (
+      <Formik
+        initialValues={{
+          signupEmail: '',
```

Comment:
why do we need the signup prefix?

Reply:
i wanted to has a prefix because ids should be unique throughout the app, so having something like `firstName` would probably show up in more than one form on the app

Reply:
why would this need to be unique throughout the app?

Reply:
following convention of classes = reusable and id = unique

https://developer.mozilla.org/en-US/docs/Web/API/Element/id

Reply:
"it must be unique in a document"
I think it's saying the id should be unique to a page not the entire app. if we have to prefix every id, our ids are going to be reallyLong.

Reply:
i think it's supposed to be the entire app? getElementById only returns one element that matches the id, where as getElementsByClass returns an array of elements